### ISSUE TITLE
Custom dimensions do not appear in Dataset schema

### ISSUE THREAD

**Issue by SebastianEdwards** [crealytics/spark-google-analytics#6]
Custom dimensions passed to the dimensions method do not appear to be added to the dataset schema as normal dimensions are. E.g.

```scala
val df = spark.read
    .format("com.crealytics.google.analytics")
    // ... auth stuff ...
    .option("startDate", "2016-11-10")
    .option("endDate", "2016-11-14")
    .option("dimensions", "dimension5,dimension6,dimension7")
    .option("queryIndividualDays", "true")
    .load()

df.select(
  col("dimension5").alias("cool dimension one"),
  col("dimension6").alias("neat dimensions two"),
  col("pageviews")
).orderBy(desc("dimension7"))

// org.apache.spark.sql.AnalysisException: cannot resolve '`dimension5`' given input columns
```

**Comment by SebastianEdwards** [comment]
It seems adding `.option("dimensions", "dimensionXX")` will expand all the dimensions into the schema. IE add `dimension1` through to `dimension20`.

However requests will then always fail with a `400 Bad Request` as there are too many dimensions in the query: `Requested 21 dimensions; only 7 are allowed.`

**Comment by nightscape** [comment]
Hi @SebastianEdwards,

I currently don't have much time to invest into this.
Would you mind debugging this by yourself a little (either stepping through the execution or adding some `println`).
It would be especially helpful to know what the [value of `allDimensions` is here](https://github.com/crealytics/spark-google-analytics/blob/master/src/main/scala/com/crealytics/google/analytics/AnalyticsRelation.scala#L85-L87) and what the full query looks like after [this line](https://github.com/crealytics/spark-google-analytics/blob/master/src/main/scala/com/crealytics/google/analytics/AnalyticsRelation.scala#L165).

Thanks!

**Comment by SebastianEdwards** [comment]
Solved by 0322d3ea46476e624e920ffee1b120ef3090859b. Thanks @nightscape!
